File: 167-two-sum-ii-input-array-is-sorted/two-sum-ii-input-array-is-sorted.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& numbers, int target) {
        int low = 0;
        int high = numbers.size() - 1;
        vector<int> solution;

        while(true) {
            // int current = numbers[low] + numbers[high];

            if (numbers[low] + numbers[high] == target) {
                solution.push_back(low + 1);
                solution.push_back(high + 1);
                        return solution;
            }
            if (numbers[low] + numbers[high] < target) {
                low++;
            } else if (numbers[low] + numbers[high] > target) {
                high--;
            }
        }
        return solution;
    }
};

File: 167-two-sum-ii-input-array-is-sorted/two-sum-ii-input-array-is-sorted.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        l, r = 0, len(numbers) - 1

        while True :
            curr_sum = numbers[l] + numbers[r]
            if (numbers[l] + numbers[r] == target):
                return [l + 1, r + 1]
            elif (curr_sum < target):
               l += 1
            else:
                r -= 1
        